subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/NativePlatformBackedFileMetadataAccessor.java

```diff
@@ -35,14 +35,24 @@ public NativePlatformBackedFileMetadataAccessor(Files files) {
 
     @Override
     public FileMetadataSnapshot stat(File f) {
-        FileInfo stat = files.stat(f, true);
+        FileInfo stat = files.stat(f, false);
+        boolean isSymlink = stat.getType() == FileInfo.Type.Symlink;
+        if (isSymlink) {
+            stat = files.stat(f, true);
+        }
```

Comment:
What happens if the symlink points to another symlink? Do we need to recursively resolve here?

Reply:
No, following symlinks also resolves chains of symlinks.